### PR TITLE
Dl.add map

### PR DESCRIPTION
**Pull request by circlegov**
Why
Allows the user to see the map of campus.

This PR
Added a constants file 
- Contains bounds for the map
- Contains a Loader with a MAP_ID

Added a map file
- Displays a map

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
+}
+
+@keyframes App-logo-spin {
+  from {
+    transform: rotate(0deg);
+  }
+  to {
+    transform: rotate(360deg);
+  }
+}
```
Comment (SofieDunt) [inline src/App.css:38]: Do we still need these styles?

File: src/components/map/index.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import React, { createRef, useEffect, useState, useCallback } from 'react';
+import { NEU_BOUNDS, LOADER } from './constants.js';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+const MapDiv = styled.div`
+  width: 100%;
+  height: 90vh; // in theory should be in a container so i can set it to 100%
```
Comment (SofieDunt) [inline src/components/map/index.js]: ❓ Should this be minHeight then? I’d also say it’s fine to set height to 100% and just leave a note on the component saying it’s required that it have a parent component with a set height. Another option would be to pass the height as a prop.

File: src/components/map/index.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    setMapElement(mapRef.current);
+  }, [mapRef]);
+
+  useEffect(() => {
+    if (mapElement) {
+      LOADER.load()
+        .then((google) => {
+          new google.maps.Map(mapElement, {
+            center: { lat: 42.33992691759904, lng: -71.08986968678391 }, // gonna hardcode for now, lat,lng
+            zoom: 16.7, // hardcode for now as well
```
Comment (SofieDunt) [inline src/components/map/index.js]: ❓Hardcode as opposed to? I’d say you might as well just pull them out into constants with the rest of the map/constants now

File: src/components/map/index.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            fullscreenControl: false,
+            mapTypeControl: false,
+            restriction: {
+              bounds: NEU_BOUNDS,
+              strictBounds: false,
+            },
+          });
+
+
+          // overlay stuff later
```
Comment (SofieDunt) [inline src/components/map/index.js]: ⛏️ (nitpick) format these with a TODO (IntelliJ/IDEs can keep track of them)

File: src/components/map/index.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+        })
+        .catch((err) => console.log(err.message));
```
Comment (SofieDunt) [inline src/components/map/index.js:47]: 📌 Out of scope for this PR, but we should consider how to notify the user that the map failed to load and what they should do (very few people will ever look at the console)

File: src/components/map/index.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    }
+  }, [mapElement, view, zoom, lat, lng, initMapCallback]);
+
+  return (
+    <>
+      <MapDiv id="map" ref={mapRef} />
+    </>
```
Comment (SofieDunt) [inline src/components/map/index.js]: You can delete the <></> tags

File: src/components/map/index.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  );
+};
+
+Map.propTypes = { 
+  view : PropTypes.any,
+  zoom : PropTypes.number,
+  lat : PropTypes.number,
+  lng : PropTypes.number,
+  initMap : PropTypes.any
```
Comment (SofieDunt) [inline src/components/map/index.js]: I think there’s a PropTypes.func that might make more sense here

Comment (SofieDunt) [reply src/components/map/index.js]: Also you might want to consider whether these props are required

File: src/components/map/index.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+  return (
+    <>
+      <MapDiv id="map" ref={mapRef} />
+    </>
+  );
+};
+
+Map.propTypes = { 
+  view : PropTypes.any,
```
Comment (SofieDunt) [inline src/components/map/index.js]: ❓What are the view types for this map?

**Review by SofieDunt — COMMENTED** [review]
🔥 Looks great! Just a couple questions and small notes

File: src/components/map/index.js (hunk 4 above)
Comment (circlegov) [reply src/components/map/index.js]: ok cool, didn't know I could do that. added.

File: src/components/map/index.js (hunk 3 above)
Comment (circlegov) [reply src/components/map/index.js]: makes sense, changed.

File: src/components/map/index.js (hunk 6 above)
Comment (circlegov) [reply src/components/map/index.js]: done.

**Review by SofieDunt — APPROVED** [review]
A couple outstanding questions, but this looks good 🔥 - let's merge it in

File: src/App.css (hunk 1 above)
Comment (circlegov) [reply src/App.css:38]: I'm not sure what this style is used for

**Pull request merged**